src/main/java/org/threadly/litesockets/NoThreadSocketExecuter.java

```diff
@@ -221,7 +224,7 @@ protected void shutdownService() {
       if(selector != null && selector.isOpen()) {
         selector.close();
       }
-    } catch (IOException e) {
+    } catch (Exception e) {
```

Comment:
you can get this stack trace if you are doing things with connection in the
selector on another thread:

java.lang.NullPointerException
	at java.nio.channels.spi.AbstractSelectableChannel.removeKey(AbstractSelectableChannel.java:128)
	at java.nio.channels.spi.AbstractSelector.deregister(AbstractSelector.java:185)
	at sun.nio.ch.EPollSelectorImpl.implClose(EPollSelectorImpl.java:147)
	at sun.nio.ch.SelectorImpl.implCloseSelector(SelectorImpl.java:114)
	at java.nio.channels.spi.AbstractSelector.close(AbstractSelector.java:108)
	at org.threadly.litesockets.NoThreadSocketExecuter.shutdownService(NoThreadSocketExecuter.java:225)

If you want I can just catch both IO and NPE, but didnt seem worth it
for just the shutdown.


On Fri, May 8, 2015 at 10:38 PM, Mike Jensen <notifications@github.com>
wrote:

> In src/main/java/org/threadly/litesockets/NoThreadSocketExecuter.java
> <https://github.com/threadly/litesockets/pull/16#discussion_r29989694>:
>
> > @@ -221,7 +224,7 @@ protected void shutdownService() {
> >        if(selector != null && selector.isOpen()) {
> >          selector.close();
> >        }
> > -    } catch (IOException e) {
> > +    } catch (Exception e) {
>
> What are you swallowing just any exception here?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/threadly/litesockets/pull/16/files#r29989694>.
>